src/praisonai/tests/unit/test_botos_drain.py

```
@@ -0,0 +1,89 @@
+Verifies that ``BotOS.drain`` quiesces ingress, waits (bounded) for
+in-flight agent turns to finish, and reports abandoned turns on timeout,
+while remaining a no-op when no drain timeout is configured.
+"""
+
+import asyncio
+
+import pytest
+
+try:
+    from praisonai.bots.botos import BotOS
+except Exception:  # pragma: no cover - optional deps not installed
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)